src/complex/DataStructure/Factory/NeighborListFactory.cpp

```diff
@@ -26,22 +26,22 @@ std::string NeighborListFactory::getDataTypeName() const
 }
 
 H5::ErrorType NeighborListFactory::readH5Group(H5::DataStructureReader& dataStructureReader, const H5::GroupReader& parentReader, const H5::GroupReader& groupReader,
-                                               const std::optional<DataObject::IdType>& parentId)
+                                               const std::optional<DataObject::IdType>& parentId, bool preflight)
 {
   return -1;
 }
 
 template <typename T>
 inline void importNeighborList(DataStructure& dataStructure, const H5::GroupReader& parentReader, const H5::DatasetReader& datasetReader, const std::string& dataArrayName, DataObject::IdType importId,
-                               const std::optional<DataObject::IdType>& parentId)
+                               const std::optional<DataObject::IdType>& parentId, bool preflight)
 {
   using NeighborListType = NeighborList<T>;
   auto dataVector = NeighborListType::ReadHdf5Data(parentReader, datasetReader);
```

Comment:
Like we talked about at the meeting, we'll need a way to read a `NeighborList` without allocating during preflight. We don't necessarily have to put it in this PR though since we want to get most of the preflight functionality in quickly.